taskcluster_util/util/downloader.py

```diff
@@ -81,8 +88,16 @@ def download_latest_artifact(self, task_id, full_filename, dest_dir):
         final_file_path = temp_local_file
         abs_dest_dir = os.path.abspath(dest_dir) if dest_dir else os.getcwd()
         try:
-            FolderHandler(abs_dest_dir).copy_elements_from(temp_local_file)
             final_file_path = os.path.join(abs_dest_dir, base_filename)
+            if is_gzip:
+                import gzip
+                logger.info('Doing gzip decode...')
+                with gzip.open(temp_local_file, 'rb') as gzip_buffer:
+                    file_content = gzip_buffer.read()
+                    with open(final_file_path, 'wb') as fd:
```

Comment:
Might need to add the folder existing check before writing the unzip file content to destination...or it will raise ERROR during execution.  ex: taskcluster_download -t Saty_TGsSA6Crq4RQChStg -a public/logs/live.log -d tmp/